pkg/chunkenc/xor.go

```diff
@@ -186,37 +188,44 @@ func (a *xorAppender) Append(t int64, vvar interface{}) {
 		a.b.writeBits(0x1f, 5)
 		a.b.writeBits(uint64(t), 51)
 		a.b.writeBits(math.Float64bits(v), 64)
-
-	} else if num == 1 {
+		a.isPreviousNewSeries = true
+	} else {
 		tDelta = uint64(t - a.t)
 
-		a.b.writeBits(tDelta, 32)
-		a.writeVDelta(v)
+		// We write time deltas as 32 bits (for compression) if the delta is too large we'll start a new series
+		if bits.Len64(tDelta) >= 32 {
```

Comment:
you can combine `num == 0` and `bits.Len64(tDelta) >= 32` into `a.hasToStartNewSeries(t)` and then reuse the code of initial block